internal/rooms/rooms.go

```diff
@@ -1576,151 +1576,86 @@ func (r *Room) FindContainerByName(containerNameSearch string) string {
 	return closeMatch
 }
 
+// FindNoun resolves a noun or alias within the room's noun map.
 func (r *Room) FindNoun(noun string) (foundNoun string, nounDescription string) {
-
 	if len(r.Nouns) == 0 {
 		return ``, ``
 	}
 
-	roomNouns := map[string]string{}
-
-	for originalNoun, originalDesc := range r.Nouns {
-		roomNouns[originalNoun] = originalDesc
-
-		if strings.Contains(originalNoun, ` `) {
-			for _, n := range strings.Split(originalNoun, ` `) {
-
-				if _, ok := r.Nouns[n]; ok {
-					continue
-				}
-
-				if _, ok := roomNouns[n]; ok {
-					continue
+	// Build flat map of noun -> desc, and add single-word aliases for multi-word nouns
+	roomNouns := make(map[string]string, len(r.Nouns))
+	for key, desc := range r.Nouns {
+		roomNouns[key] = desc
+		if strings.Contains(key, ` `) {
+			for word := range strings.SplitSeq(key, " ") {
+				if _, exists := r.Nouns[word]; !exists && roomNouns[word] == "" {
+					roomNouns[word] = `:` + key
 				}
-
-				roomNouns[n] = `:` + originalNoun
-
 			}
 		}
-
 	}
 
-	testNouns := util.SplitButRespectQuotes(noun)
-	ct := len(testNouns)
-	for i := 0; i < ct; i++ {
-
-		if splitCount := strings.Split(testNouns[i], ` `); len(splitCount) > 1 {
-
-			for _, n2 := range splitCount {
-
-				if len(n2) < 2 {
-					continue
-				}
-
-				testNouns = append(testNouns, n2)
-			}
-
+	// Build candidate list: full noun first, then each word
+	candidates := []string{noun}
+	for _, part := range util.SplitButRespectQuotes(noun) {
+		if len(part) > 1 {
+			candidates = append(candidates, part)
 		}
 	}
 
-	// If it created more than one word, put the original back on as a full string to test
-	if len(testNouns) > 1 {
-		testNouns = append(testNouns, noun)
-	}
-
-	for _, newNoun := range testNouns {
-
-		if desc, ok := roomNouns[newNoun]; ok {
-			if desc[0:1] == `:` {
-				return desc[1:], roomNouns[desc[1:]]
+	// Try direct matches and aliases
+	for _, cand := range candidates {
+		if desc, ok := roomNouns[cand]; ok {
+			key := cand
+			for strings.HasPrefix(desc, `:`) {
+				key = desc[1:]
+				desc = roomNouns[key]
 			}
-			return newNoun, desc
-		}
-
-		if len(newNoun) < 2 {
-			continue
+			return key, desc
 		}
+	}
 
-		// If ended in `s`, strip it and add a new word to the search list
-		if newNoun[len(newNoun)-1:] == `s` {
-
-			testNoun := newNoun[:len(newNoun)-1]
-			if desc, ok := roomNouns[testNoun]; ok {
-				if desc[0:1] == `:` {
-					return desc[1:], roomNouns[desc[1:]]
-				}
-				return testNoun, desc
-			}
-
-		} else {
-
-			testNoun := newNoun + `s`
-			if desc, ok := roomNouns[testNoun]; ok { // `s`` at end
-				if desc[0:1] == `:` {
-					return desc[1:], roomNouns[desc[1:]]
+	// Fallback pluralization/singularization
+	for _, cand := range candidates {
+		// Handle "ies" -> "y"
+		if strings.HasSuffix(cand, `ies`) {
```

Comment:
Thoughts on a `Pluralize()` function that returns the plural form of a string? This would let us do a couple of things:

1. Add as many plurals as we want with a clear list (can be an array of defined suffixes/replacements)
2. Take language/localization into consideration.

Not a requirement for this, just a thought. I might create a separate PR with a pluralization feature added to the language package.

Reply:
I have using this package for Pluralize just as a fyi

Reply:
That looks pretty good... doesn't seem to support languages, but same general idea. Also pluralize and singularize I guess is what we need.

Just ideas right now. Luckily these aren't used for display, just some extra attempts to match what they typed with what nouns are available, so being fast and loose with it is fine.